acme/challenge.go

```diff
@@ -120,6 +128,36 @@ func (c Challenge) DNS01KeyAuthorization() string {
 	return base64.RawURLEncoding.EncodeToString(h[:])
 }
 
+// MailReply00KeyAuthorization encodes a key authorization value to be sent
+// back to acme challange's reply-to address.
+// Subject of that mail is saparate token.
+// RFC8823 §3.1
+func (c Challenge) MailReply00KeyAuthorization(mailsubject string) string {
+	//if subject given has ACME: header, strip it before calculate keyauth
+	mailsubject = strings.TrimPrefix(mailsubject, "ACME: ")
+	h := sha256.Sum256([]byte(mailsubject + c.KeyAuthorization))
+	return base64.RawURLEncoding.EncodeToString(h[:])
+}
+
+// MailChallangeReplyGen builds email body with headers to reply MailReply00
+// challange Response email. This fucntion just build body, and sendung
+// message have to done by caller of this function
+
+func (c Challenge) MailChallangeReplyGen(mailsubject string, messgageid string) string {
+	mailsubject = strings.TrimPrefix(mailsubject, "ACME: ")
+	keyauth := c.MailReply00KeyAuthorization(mailsubject)
+	msg := fmt.Sprint("To: ", c.Identifier.Value, "\r\n",
+		"In-Reply-To: ", c.From, "\r\n",
+		"Subject: RE: ACME: ", mailsubject, "\r\n",
+		"Content-Type: text/plain\r\n",
+		"\r\n",
+		"-----BEGIN ACME RESPONSE-----\r\n",
+		keyauth, "\r\n",
+		"-----END ACME RESPONSE-----\r\n",
+	)
+	return msg
+}
```

Comment:
Thanks for the helper functions! I think these would be a better fit in the higher-level `acmez` package, rather than the `acme` package. (And should be functions, not methods on Challenge -- but they'd take a Challenge as an argument, much like the [TLSALPN01ChallengeCert function](https://pkg.go.dev/github.com/mholt/acmez#TLSALPN01ChallengeCert)).

One more question: what/who actually sets the value of the `From` field?